source/Cosmos.System2_Plugs/System/IO/FileImpl.cs

```diff
@@ -162,11 +162,46 @@ public static void WriteAllBytes(string aFile, byte[] aBytes)
 
         public static void Copy(string srcFile, string destFile)
         {
-            using (var xFS = new FileStream(srcFile, FileMode.Open))
+            try
             {
-                var xBuff = new byte[(int)xFS.Length];
-                var yFS = new FileStream(destFile, FileMode.Create);
-                yFS.Write(xBuff, 0, xBuff.Length);
+                byte[] srcFileBytes = File.ReadAllBytes(srcFile);
+                File.WriteAllBytes(destFile, srcFileBytes);
+            }
+            catch (IOException ioEx)
+            {
+                throw new IOException("File Copy", ioEx);
+            }
+        }
+
+        public static void Copy(string srcFile, string destFile, bool overwriting)
+        {
+            if (overwriting)
+            {
+                File.Delete(destFile);
+                try
+                {
+                    byte[] srcFileBytes = File.ReadAllBytes(srcFile);
+                    File.WriteAllBytes(destFile, srcFileBytes);
+                }
+                catch (IOException ioEx)
+                {
+                    throw new IOException("File Copy", ioEx);
+                }
+            }
+            else
+            {
+                if (!File.Exists(destFile))
```

Comment:
If the file exists, an exception should be thrown. From [MSDN](https://msdn.microsoft.com/en-us/library/9706cfs5(v=vs.110).aspx):

|IOException | destFileName exists and overwrite is false. -or- An I/O error has occurred. 
| -- | --